flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java

```diff
@@ -82,7 +73,7 @@
 
     private static final long serialVersionUID = 6642514795175288193L;
 
-    private static final String NEXT_LEFT_INDEX_STATE_NAME = "next-index";
+    // TODO: do we have to rename states so Flink fails early when someone tries to restore from old state?
```

Comment:
No, table api doesn't support upgrades yet :)

Reply:
question is how well it fails if we keep these names - are you confident that I can just remove the comment?